.github/workflows/check.yml

```diff
@@ -44,41 +31,43 @@ jobs:
     runs-on: ubuntu-latest
     container:
       image: docker://rust:1.56.0-buster
-    permissions:
-      contents: read
     steps:
       - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
-      - run: for d in $(for toml in $(find . -mindepth 2 -name Cargo.toml -not -path '*/fuzz/*') ; do echo ${toml%/*} ; done | sort -r ) ; do echo "# $d" ; (cd $d ; cargo check --all-targets) ; done
+      - run: |
+          for d in $(for toml in $(find . -mindepth 2 -name Cargo.toml -not -path '*/fuzz/*') ; do echo ${toml%/*} ; done | sort -r ) ; do
+            echo "# $d"
+            (cd $d ; cargo check --all-targets)
+          done
 
   # Enforce automated formatting.
   fmt:
-    timeout-minutes: 5
+    timeout-minutes: 10
     runs-on: ubuntu-latest
     container:
       image: docker://rust:1.56.0-buster
-    permissions:
-      contents: read
     steps:
       - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
       - run: rustup component add rustfmt
       - run: make check-fmt
 
-  # Run all tests.
-  test:
+  # Generate docs
+  docs:
     timeout-minutes: 15
     runs-on: ubuntu-latest
     permissions:
       contents: read
     steps:
       - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
-      - run: make test
+      - run: cargo doc --all
 
-  # Generate docs
-  docs:
-    timeout-minutes: 15
+  # It's easy to make changes that are innocuous in dev builds that end up ballooning resources
+  # needed for release builds. This job builds the proxy in release-mode to ensure the build isn't
+  # OOM-killed.
+  release-binary:
+    timeout-minutes: 20
     runs-on: ubuntu-latest
-    permissions:
-      contents: read
     steps:
       - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
-      - run: cargo doc
+      - env:
+          CARGO_RELEASE: true
+        run: make build
```

Comment:
since the release build could potentially take way longer than the other steps, should this go in a separate workflow?